compiler/rustc_mir/src/borrow_check/diagnostics/conflict_errors.rs

```diff
@@ -216,12 +216,14 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                                 );
                             }
                             // Avoid pointing to the same function in multiple different
-                            // error messages
-                            if self.fn_self_span_reported.insert(self_arg.span) {
-                                err.span_note(
-                                    self_arg.span,
-                                    &format!("this function takes ownership of the receiver `self`, which moves {}", place_name)
-                                );
+                            // error messages.
+                            if span != DUMMY_SP {
+                                if self.fn_self_span_reported.insert(self_arg.span) {
```

Comment:
Can we make it a single `if` to avoid rightwards drift?

```suggestion
                            if span != DUMMY_SP && self.fn_self_span_reported.insert(self_arg.span) {
```